src/components/Header/Header.jsx

```diff
@@ -1,18 +1,32 @@
 import React from "react";
 import { useHistory } from "react-router-dom";
+import { useTermContext } from "../../utils/contexts/TermContext";
 import "./Header.styles.css";
 
 function Header() {
   const history = useHistory();
+  const { setTerm } = useTermContext();
 
   const redirectToHome = () => history.push("/home");
-  
+
+  const handleKeyPressed = (event) => {
+    const term = event.target.value;
+    if (event.key === "Enter" && term.trim() !== "") {
+      setTerm(term);
+      history.push(`/search?q=${term}`);
+    }
+  };
+
   return (
     <div className="header">
       <label onClick={redirectToHome} className="logo">
         Wize Home
       </label>
-      <input className="search" placeholder="Search..." />
+      <input
+        className="search"
+        onKeyPress={(e) => handleKeyPressed(e)}
```

Comment:
`handleKeyPressed` is already a function that requires the same params as the arrow function defined in the `onKeyPress`, the arrow function can be deleted:

```suggestion
        onKeyPress={handleKeyPressed}
```